vignettes/function-flow.Rmd

```diff
@@ -95,63 +95,202 @@ outputs.](images/function-flow-population.png)
-extract all events of tests above 48 mmol/mol.
+extract the dates of all elevated HbA1c test results: $\geq$ 48 mmol/mol
+(IFCC units, `analysiscode` NPU27300) or $\geq$ 6.5% (DCCT units,
+`analysiscode` NPU03835). Multiple elevated results on the same day
+within each individual are deduplicated, to account for the same test
+result being reported as two rows (one for IFCC, one for DCCT units).
 
-<!-- TODO: Add details on how this filtering should be done -->
+`include_hba1c()` outputs a 2-column data frame containing the following
+variables:
+
+-   identifier variable (`pnr`)
+-   the dates of all elevated HbA1c test results (`do_pos_hba1c`).
+
+The output is passed to the `exclude_pregnancy()` function for censoring
+of elevated results due to potential gestational diabetes (see below).
 
 #### Hospital diagnosis of diabetes
 
-The function `include_diabetes_diagnoses()` uses the hospital contacts
-from LPR2 and 3 to include all dates of diabetes diagnoses. Diabetes
-diagnoses from both ICD 8 and ICD 10 are included.
+**Joining LPR2 and LPR3 data**
 
-This function contains two helper functions:
+The helper functions `join_lpr2()` and `join_lpr3()` join records of
+diagnoses to administrative information in LPR2-formatted and
+LPR3-formatted data, respectively.
+
+`join_lpr2()` takes `lpr_diag` and `lpr_adm` as inputs, joins the
+required information by record number (`recnum`), and outputs a
+`data.frame` with the following variables:
+
+-   identifier variable (`pnr`)
+-   date (`d_inddto`)
+-   department specialty (`c_spec`)
+-   diagnosis code (`c_diag`)
+-   diagnosis type (`c_diagtype`)
+
+`join_lpr3()` takes `diagnoser` and `kontakter` as inputs, joins the
+required information by record number (`dw_ek_kontakt`), and outputs a
+`data.frame` with the following variables:
 
--   `keep_diabetes_icd10()`
--   `keep_diabetes_icd8()`
+-   identifier variable (`cpr`)
+-   date (`dato_start`)
+-   department specialty (`hovedspeciale_ans`)
+-   diagnosis code (`diagnosekode`)
+-   diagnosis type (`diagnosetype`)
+-   diagnosis retracted (`senere_afkraeftet`)
 
-<!-- TODO: Add details on how this filtering should be done, e.g., diagnosis codes -->
+These outputs are passed to `include_diabetes_diagnoses()` for further
+processing, see below.
 
-<!-- TODO: Which specific ICD 8 and 10 codes are included? -->
+**Processing of diagnoses**
+
+The function `include_diabetes_diagnoses()` uses the hospital contacts
+from LPR2 and LPR3 to include all dates of diabetes diagnoses to use for
+inclusion, as well as additional information needed to classify diabetes
+type. Diabetes diagnoses from both ICD-8 and ICD-10 are included.
+
+The function takes the outputs of `join_lpr2()` and `join_lpr3()` as
+inputs and processes each input separately to generate the following
+internal variables:
+
+-   LPR2-data:
+    -   `pnr`: identifier variable
+    -   `do_diagnosis`: include all diabetes diagnoses, registered as
```

Comment:
What's the `do_` mean?

Reply:
date of?

Reply:
Yes. It's ugly and I'm open to changing it... maybe diagnosis_date? (the only thing to keep in mind is that we should maybe name the variable in way that doesn't get it confused with the algorithm's classified date of diabetes onset - if it's needed/possible?

Reply:
Since it is an internal function, calling it `diagnosis_date` I think would be fine.

Reply:
I went and simply called it "date". If it's too close to the "dates" variable output to the join_inclusions() function, we can change it.